lib/guess_html_encoding.rb

```diff
@@ -19,11 +19,20 @@ def self.guess(html, headers = nil)
     end
 
     if out.nil? || out.empty? || !encoding_loaded?(out)
-      if html =~ /<meta[^>]*HTTP-EQUIV=["']?Content-Type["']?[^>]*content=["']([^'"]*)["']/i && $1 =~ /charset=([\w\d-]+);?/i
-        out = $1
-      elsif html =~ /<meta\s+charset=["']([\w\d-]+)?/i
-        out = $1
+
+      meta_tags_regex = /
+      (?:
+        <meta\s*HTTP-EQUIV=["']?Content-Type["']?[^>]*content=["'][^'"]*charset\s*=\s*([\w\d-]+)[;"']  # http-equiv
```

Comment:
This regex changed a bit.  Did you intend to force one of `[;"']` at the end? What if they just close the tag?

Also, at the beginning, we lost `[^>]*` which would take an optional additional argument to the `meta`, although that isn't necessary.